### DISCUSSION TITLE
Setup.html

### DISCUSSION THREAD

**Discussion by Baatezu** [guydavis/machinaris#768] [General]
In this field when we put 24 words, browser keep this words, im not really understand if we can block it on "our side"

```
<form id="main-form" method="POST">
                            <input class="form-control me-2" id="mnemonic" name="mnemonic" type="text"
                                placeholder="north divorce holiday whisper inquiry toy rare minute father ostrich surprise runway ill episode dance rather unaware concert sunset maid gather grain gift wire"
                                aria-label="Import Key">
                            <br />
                            <button class="btn btn-outline-success" type="submit" id="btnImport">{{_('Import Existing Key')}}</button>
                            <h3 style="padding-top: 25px;">{{_('OR')}}</h3>
```

**Comment by guydavis** [comment]
Hi, thanks for the question.  Would you mind elaborating on your concern?  I don't follow when you say: 

 `im not really understand if we can block it on "our side"`

Here is link to the [relevant wiki page](https://github.com/guydavis/machinaris/wiki/Keys#setup).  The `placeholder` in the html code you pasted above is displayed as a example in the Import Key input field.  This [placeholder](https://www.w3schools.com/tags/att_input_placeholder.asp) text vanishes as soon as you click into the input fiield and start typing (or paste the key to import).

Thanks,
Guy

**Reply by Baatezu** [reply]
try to add little secure. when I reinstall machinaris and type mnemonic I will surprise browser has it already :)

if add:  autocomplete="off" browser stop "cache" mnemonic on setup.html page?

like this: <input class="form-control me-2" id="mnemonic" name="mnemonic" type="text" **autocomplete="off"**

**Reply by guydavis** [reply]
Ahh! I see now.  Yes, in the case where you perform a setup once, importing your mnemonic, then later return to the [Setup page](https://github.com/guydavis/machinaris/wiki/Keys#setup) for a re-install, the browser would redisplay the key value, cached client-side only.  

Great suggestion to disable this browser `autocomplete` behavior for this form field!  I've just [committed this change](https://github.com/guydavis/machinaris/commit/500e070e3c23d73eabf862d9439665ccb8594e20).  Thanks again for this valuable feedback.  Please me know if you have any other questions or suggestions about Machinaris.
